numba_dppy/dpctl_iface/dpctl_capi_fn_builder.py

```diff
@@ -295,3 +295,38 @@ def get_dpctl_event_delete(builder, context):
             arg_list=[void_ptr_t],
             func_name="DPCTLEvent_Delete",
         )
+
+    @staticmethod
+    def get_dpctl_queue_memset(builder, context):
+        """
+        get_dpctl_queue_memset(builder, context)
+        Inserts an LLVM Function for the ``DPCTLQueue_Memset``.
+
+        The ``DPCTLQueue_Memset`` function is a wrapper over ``sycl::queue``
```

Comment:
dpctl currently does not expose a function `DPCTLQueue_Memset` function. It needs to be first added to dpctl's `libsyclinterface` library. Can you open a PR against dpctl as well? A while back dpctl memory functions where changes to return event refs and the Memset wrapper will follow the same design (in fact the signature you have in the docstring does return an event). Therefore, remove the comment about "the DPCTLQueue_Memset does not return an event reference".

Reply:
Yes, I made a request to dpctl for memset implementation.

Reply:
I meant can you open a PR providing the implementation. @1e-to can you help as you have lots of experience with libsyclinterface.

Reply:
The `DPCTLQueue_Memset` has been added in IntelPython/dpctl#812 and exposed via Python API in IntelPython/dpctl#815.